src/crab/ebpf_domain.hpp

```
@@ -66,19 +73,34 @@ class EbpfDomain final {
+    ExtendedNumber get_loop_count_upper_bound(std::span<const Variable> loop_counters) const;
     Interval get_r0() const;
 
-    static EbpfDomain setup_entry(bool init_r1);
-    static EbpfDomain from_constraints(const std::set<std::string>& constraints, bool setup_constraints);
+    static EbpfDomain setup_entry(bool init_r1, const AnalysisContext& context);
+    static EbpfDomain from_constraints(const std::set<std::string>& constraints, bool setup_constraints,
+                                       const AnalysisContext& context);
+    /// Direct construction from typed constraints. The stack remains top at the
+    /// requested size; pure-semantics callers (tests) can omit it for the
+    /// default-options size.
     static EbpfDomain from_constraints(const std::vector<std::pair<Variable, TypeSet>>& type_restrictions,
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)